src/Gulpfile.js

```diff
@@ -237,7 +237,7 @@ function buildJsPipeline(assetGroup, doConcat, doRebuild) {
         ))
         .pipe(plumber())
         .pipe(gulpif(generateSourceMaps, sourcemaps.init()))
-        .pipe(typescript(typeScriptOptions))
+        .pipe(typescript(typeScriptOptions)) // typescript does not support javascript modules
```

Comment:
Typescript should support javascript modules out of the box. I would need to see an example of what you are refering to.

Reply:
I am talking specifically about gulp-typescript here, because I'm not familiar enough with typescript to make blanket statements.
Typescript does support javascript modules, as long as there also are other dependencies. I mean to say that the transpilation step for modules that are written in .js files results in javascript that depends on other libraries.
For example, it could create javascript that depends on system.js, or amd.js, but doesn't seem to be able to produce javascript that stands on its own. The specific library the js will depend on is "selected" by a property in typeScriptOptions, but the library itself is not automatically included (it would be hard to do so at this stage), so the end result is that the resulting js fails to execute.

Reply:
Have you tried updating gulp-typescript?
"gulp-typescript": "3.1.4",

Because as you can see here :  

https://github.com/OrchardCMS/OrchardCore/blob/5310bdb49797e9db02f3107ea9c0feb270e92346/package.json#L54

We are using a newer version. And also, when you say that it produces a non-working javascript file maybe that's because we were back then using a lower version of TypeScript too. So, one thing that could be done here is to try updating TypeScript to the latest version and try to transpile manually these modules assets. Then, if it works, update the gulp-typescript version to the latest version too and see if that works.

Reply:
Basically it should support TypeScript 2.2.0

https://github.com/ivogabe/gulp-typescript/pull/457/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R83